### PR TITLE
Latest Episodes!

### PR DESCRIPTION
**Pull request by yougotwill**
Hi, been a while since my last PR but I had been working on a few small fixes on a personal branch since I still can't build the `dev` branch. Aside from those small fixes (not in this PR) this is a big feature that I wanted to add for a while.

### Introducing Latest Episodes!

Screenshot 2020-06-13 at 11 22 14

Basically Poddr will fetch the most recent episode from each favourited podcast and display it to the user. It speeds up my navigation experience a lot!

It's been tested on my machine using a local branch based on `master`. I just copied those files to a new branch based on `dev` for this PR but everything should work.

I'm new to Angular and Typescript so if there is any code refactoring that needs to be done please let me know :)

### CODE REVIEW

**Comment by yougotwill** [comment]
Still need to order episodes in descending order by published

**Comment by Sn8z** [comment]
Hey @yougotwill 
Sorry for taking so long to answer, my apologies!

This looks good but I'm actually working on something similar that's not yet been pushed to the repo (my bad..) so I won't merge this but I might take some ideas and inspiration from your hard work :)

Code looks fine but there are some minor things I would refactor, I'll leave some comments in this PR if you are interested ^^

File: src/app/latest/latest.component.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  public faList = faList;
+
+  constructor(private podcastService: PodcastService,
+    private favService: FavouritesService,
+    private toast: ToastService,
+    private descriptionPipe: Description) { }
+
+  ngOnInit() {
+    this.layout = this.store.get("layout", "grid") as string;
+    this.favService.favourites.subscribe(value => {
```
Comment (Sn8z) [inline src/app/latest/latest.component.ts:32]: I realize I've made the same choice in my code but it should probably be a more descriptive variable name than 'value' :)

File: src/app/sidenav/sidenav.component.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Component, OnInit } from '@angular/core';
 import { AudioService } from '../services/audio.service';
-import { faFire, faSearch, faTools } from '@fortawesome/free-solid-svg-icons';
+import { faFire, faBolt faSearch, faTools } from '@fortawesome/free-solid-svg-icons';
```
Comment (Sn8z) [inline src/app/sidenav/sidenav.component.ts:3]: Here's a missing comma but those things happen when you can't build the project so it's understandable!

File: src/app/sidenav/sidenav.component.ts [hunk 3]
```diff
@@ -11,8 +11,9 @@ import { faHeart } from '@fortawesome/free-regular-svg-icons';
 export class SidenavComponent implements OnInit {
   public image: string = "";
   public rss: string;
-	
+
 	public faFire = faFire;
+  public faBolt = faBolt;
```
Comment (Sn8z) [inline src/app/sidenav/sidenav.component.ts:16]: Indentation is a bit weird.
I've an ambition to create a styleguide sometime in the future but no ETA for that yet.

**Comment by Sn8z** [comment]
A few other notes:
- The CSS is mostly duplicated which means it should be moved to a common CSS file instead to avoid the need to edit multiple files for the same style change to take effect.
- When adding a completely new component there are more things to consider, for example shortcuts.

Even though this won't be merged I'm really grateful for your work and dedication :man_dancing:

**Comment by yougotwill** [comment]
> Hey @yougotwill
> Sorry for taking so long to answer, my apologies!
> 
> This looks good but I'm actually working on something similar that's not yet been pushed to the repo (my bad..) so I won't merge this but I might take some ideas and inspiration from your hard work :)
> 
> Code looks fine but there are some minor things I would refactor, I'll leave some comments in this PR if you are interested ^^

Hey @Sn8z no problem! Glad to hear you are working on something similar. Looking forward to the release ;) I'm treating my contributions here as a learning exercise so thank you for the feedback it is much appreciated!

**Pull request closed without merging**